pkg/scheduler/queue/tenant_queues.go

```diff
@@ -99,23 +90,12 @@ func (qb *queueBroker) enqueueRequestBack(request *tenantRequest, tenantMaxQueri
 		return err
 	}
 
-	// TODO (casie): When deprecating TreeQueue, clean this up.
-	// Technically, the MultiQueuingAlgorithmTreeQueue approach is adequate for both tree types, but we are temporarily
-	// maintaining the legacy tree behavior as much as possible for stability reasons.
-	if tq, ok := qb.tree.(*TreeQueue); ok {
-		if tenantQueueNode := tq.getNode(queuePath[:1]); tenantQueueNode != nil {
-			if tenantQueueNode.ItemCount()+1 > qb.maxTenantQueueSize {
-				return ErrTooManyRequests
-			}
-		}
-	} else if _, ok := qb.tree.(*MultiQueuingAlgorithmTreeQueue); ok {
-		itemCount := 0
-		for _, tenantNode := range qb.tenantQuerierAssignments.tenantNodes[string(request.tenantID)] {
-			itemCount += tenantNode.ItemCount()
-		}
-		if itemCount+1 > qb.maxTenantQueueSize {
-			return ErrTooManyRequests
-		}
+	itemCount := 0
+	for _, tenantNode := range qb.tenantQuerierAssignments.tenantNodes[string(request.tenantID)] {
+		itemCount += tenantNode.ItemCount()
+	}
+	if itemCount+1 > qb.maxTenantQueueSize {
```

Comment:
`if itemCount >= qb.maxTenantQueuSize` ?

Reply:
The (+1) is the request we want to enqueue -- if enqueuing it would exceed the max tenant queue size, we should not enqueue it.

Reply:
It's just a simpler way of expressing the same thing (I think?) 
`if itemCount == qb.maxTenantQueueSize` means "is the queue is already full up, we cannot add anymore" - I just added `>=` for safety.

"Current item count plus one is greater than the limit" vs. "Current item count is already at the limit"

Reply:
Oh I see what you mean; I guess it's up for debate which is easier to understand, I hadn't actually changed any logic here (the change is an indent since I got rid of a type assertion).